craft_parts/overlays/layers.py

```diff
@@ -98,3 +101,51 @@ def save(self, part: Part) -> None:
     def hex(self) -> str:
         """Return the current hash value as a hexadecimal string."""
         return self.digest.hex()
+
+
+class LayerStateManager:
+    """An in-memory layer state management helper for action planning.
+
+    :param part_list: The list of parts in the project.
+    """
+
+    def __init__(self, part_list: List[Part], base_layer_hash: Optional[LayerHash]):
+        self._part_list = part_list
+        self._base_layer_hash = base_layer_hash
+
+        self._layer_hash: Dict[str, Optional[LayerHash]] = dict()
+        for part in part_list:
+            self.set_layer_hash(part, LayerHash.load(part))
+
+    def get_layer_hash(self, part: Part) -> Optional[LayerHash]:
```

Comment:
why not require part_name instead

Reply:
Indeed, this could be the part name. Changing to use it instead.